src/components/templates/Journals/Journals.tsx

```diff
@@ -1,48 +1,99 @@
 import { useRouter } from 'next/router';
-import { useCallback } from 'react';
+import { useCallback, useState } from 'react';
 
 import { Card } from '@components/Card';
-import { Text } from '@components/Text';
+import { Container } from '@components/Container';
+import { EllipsisText } from '@components/EllipsisText';
 import { UserJournal } from '@defs/journals';
+import { DeleteOutline, EditOutline, ViewOutline } from '@icons';
+import { TravelLogsActions } from '@templates/TravelLogs';
 
 export type JournalsProps = {
-  journal?: UserJournal;
+  journal: UserJournal;
+  onUpdate: () => void;
+  onDelete: () => void;
 };
-function Journals({ journal }: JournalsProps) {
+function Journals({ journal, onUpdate, onDelete }: JournalsProps) {
   const router = useRouter();
   const onClick = useCallback(() => {
-    if (journal) {
-      router.push(`/${journal.journalId}`).catch(console.error);
-    } else {
-      // TODO: Create a new journal
-    }
-  }, []);
+    router
+      .push(`/${journal.journalId}`)
+      .catch((err) => console.error('Redirect to journal failed', err));
+  }, [journal.journalId, router]);
+
+  const [isActionVisible, setIsActionVisible] = useState(false);
+
   return (
     <Card
       isPressable
       css={{
-        bg: journal ? '$primary' : '$accents8',
-        height: '200px',
-        alignItems: 'center',
-        p: '$md',
-        '@xs': { height: '300px', p: '$xl' },
-        '@md': { height: '400px', p: '$xl' },
+        display: 'flex',
+        flexDirection: 'column',
+        overflow: 'hidden',
+        bg: '$primary',
+        height: '515px',
+        width: '406px',
+        alignItems: 'flex-start',
+        borderRadius: '$none',
+        borderTopRightRadius: '$xxl',
+        borderBottomRightRadius: '$xxl',
       }}
+      onMouseLeave={() => setIsActionVisible(false)}
+      onMouseOver={() => setIsActionVisible(true)}
       onPress={() => onClick()}
     >
-      {journal ? (
-        <>
-          <Text h4 css={{ textAlign: 'center', mb: '$sm' }}>
-            {journal.title}
-          </Text>
+      <Container
+        css={{
+          overflow: 'hidden',
+          alignItems: 'flex-start',
+          p: '$xl',
+          pl: '$3xl',
+          gap: '$xl',
+          flex: 1,
+        }}
+      >
+        <EllipsisText
+          uppercase
+          color="$onPrimary"
+          containerStyle={{
+            maxHeight: '80px',
+            overflow: 'hidden',
+            mb: '$sm',
+          }}
+          css={{
+            overflow: 'hidden',
```

Comment:
`overflow: hidden` can be set in EllipsisText itself. It's likely the default behaviour required. Correct?